### PR TITLE
Login

### PR DESCRIPTION
**Pull request by RadwaMohamed29**

### CODE REVIEW

File: app/src/main/java/com/example/news/data/local/Dao.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ import com.example.news.model.User
 interface Dao {
     @Insert
     suspend fun insert(user: User)
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/data/local/Dao.kt:11]: do you want to think about conflict strategy?

File: app/src/main/java/com/example/news/data/local/Dao.kt [hunk 2]
```diff
@@ -9,6 +9,7 @@ import com.example.news.model.User
...
+
     @Query("SELECT EXISTS (SELECT * FROM users WHERE email =:email AND password =:password)")
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/data/local/Dao.kt:13]: bravo!

File: app/src/main/java/com/example/news/data/local/Dao.kt [hunk 3]
```diff
@@ -9,6 +9,7 @@ import com.example.news.model.User
...
     fun getUser (email :String , password:String):Boolean
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/data/local/Dao.kt:14]: maybe change the function name?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -6,4 +6,8 @@
     <string name="password">Password</string>
     <string name="phone_number">Phone Number</string>
     <string name="register">Register</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (AbdulrahmanAli19) [inline app/src/main/res/values/strings.xml:9]: Unneeded tag

File: app/src/main/java/com/example/news/ui/login/view/LoginFragment.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import androidx.navigation.Navigation
+import com.example.news.databinding.FragmentLoginBinding
+import com.example.news.model.UserRepository
+import com.example.news.ui.login.viewModel.LoginViewModel
+import com.example.news.ui.login.viewModel.LoginViewModelFactory
+
+
+class LoginFragment : Fragment() {
+    lateinit var binding: FragmentLoginBinding
+    lateinit var navController: NavController
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/ui/login/view/LoginFragment.kt:19]: you can use `findNavController` ktx function.

File: app/src/main/java/com/example/news/ui/login/view/LoginFragment.kt [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentLoginBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.mutableLiveData.observe(viewLifecycleOwner) {
+            it.let {
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/ui/login/view/LoginFragment.kt:39]: you can do this the annoyingly MVVM pure way.

File: app/src/main/java/com/example/news/ui/login/viewModel/LoginViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.news.model.UserRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class LoginViewModel(private val userRepository: UserRepository) : ViewModel() {
+    var mutableLiveData=MutableLiveData<Boolean>()
```
Comment (RemonShehata) [inline app/src/main/java/com/example/news/ui/login/viewModel/LoginViewModel.kt:11]: don't expose mutable live data to the client.

**Review by RemonShehata — CHANGES_REQUESTED** [review]

**Pull request merged**